From Thomas.Koenig@ciw.uni-karlsruhe.de Sat Aug 20 11:34:02 EDT 1994
Article: 2607 of comp.os.linux.announce
Newsgroups: comp.os.linux.announce
Path: bigblue.oit.unc.edu!concert!news-feed-1.peachnet.edu!news.duke.edu!zombie.ncsc.mil!MathWorks.Com!news.kei.com!travelers.mail.cornell.edu!cornell!bounce-bounce
From: ig25@fg70.rz.uni-karlsruhe.de (Thomas Koenig)
Subject: nfsd bug and fix
Message-ID: <ann-10345.777397799@cs.cornell.edu>
Followup-To: comp.os.linux.misc
Keywords: nfs, bug report, patch
Sender: mdw@cs.cornell.edu (Matt Welsh)
Reply-To: Thomas.Koenig@ciw.uni-karlsruhe.de
Organization: University of Karlsruhe, Germany
Date: Sat, 20 Aug 1994 15:50:13 GMT
Approved: linux-announce@tc.cornell.edu (Matt Welsh)
Lines: 36

Hi;

there's a bug in the nfs suite 1.5.10 (for example on

/sunsite.unc.edu:pub/Linux/system/Network/aris/sources/nfsd/nfsd-1.5.10.tar.gz

which prevents a directory from being exported to different domains.

Consider a line in /etc/export:

/export		*.foo.do.main(ro) *.bar.another.domain(ro)

This would only allow importing the file system /export from
machines from *.bar.another.domain.

The patch to fix this, one forgotten continue, follows:

--- nfsd-1.5.10-old/auth_clnt.c	Fri Nov  5 04:18:43 1993
+++ nfsd-1.5.10/auth_clnt.c	Wed Aug 17 22:53:12 1994
@@ -153,6 +153,7 @@
 				goto found_it;
 			}
 			cpp = &(cp->next);	/* the normal iteration	*/
+			continue;
 		}
 		else if ((hent = gethostbyname(cp->clnt_name)) != NULL) {
 			cp->clnt_addr = *((struct in_addr *) hent->h_addr);
--
Thomas Koenig, Thomas.Koenig@ciw.uni-karlsruhe.de, ig25@dkauni2.bitnet.
The joy of engineering is to find a straight line on a double
logarithmic diagram.


--
Send submissions for comp.os.linux.announce to: linux-announce@tc.cornell.edu
Be sure to include Keywords: and a short description of your software.


